integration/query_frontend_cache_test.go

```diff
@@ -173,9 +183,34 @@ func generateExpectedValues(start time.Time, end time.Time, step time.Duration,
 
 	expected := model.Matrix{
 		&model.SampleStream{
-			Metric: map[model.LabelName]model.LabelValue{"__name__": "series_1"},
+			Metric: map[model.LabelName]model.LabelValue{"__name__": model.LabelValue(seriesName)},
 			Values: values,
 		},
 	}
 	return expected
 }
+
+func generateExpectedHistograms(start time.Time, end time.Time, step time.Duration, sampleTime time.Time, expectedVector model.Vector, seriesName string) model.Matrix {
+	hist := expectedVector[0].Histogram
+
+	const loopbackPeriod = 5 * time.Minute
+
+	histograms := []model.SampleHistogramPair(nil)
+	for ts := start; !ts.After(end); ts = ts.Add(step) {
+		if ts.Before(sampleTime) || ts.After(sampleTime.Add(loopbackPeriod)) {
```

Comment:
> should we not ignore timestamp outside `[sampleTime-5m, sampleTime]`?

It's actually `[sampleTime, sampleTime+5m]`. Before sample existed (sampleTime), it can't be included in the result. And it's included in the result up to 5 minutes after sample timestamp.

Reply:
Oh I see, yeah. Confused sample time with query time.